app/src/main/java/org/mozilla/fenix/components/history/PagedHistoryProvider.kt

```diff
@@ -200,14 +215,34 @@ class DefaultPagedHistoryProvider(
         // url, but we don't have a use case for this currently in the history view.
         result.addAll(
             historyGroupsInOffset.map { group ->
-                group.copy(items = group.items.distinctBy { it.url }.filterNot { redirectsInThePage.contains(it.url) })
+                group.copy(
+                    items = group.items.distinctBy { it.url }
+                        .filterNot { redirectsInThePage.contains(it.url) }
+                        .filter { shouldShowUrl(historyHighlights, it.url, it.visitedAt) }
```

Comment:
Adding filter using highlights as a easy way since highlights doesn't have `visitedAt` value

Reply:
As discuss, remove the filter here.